python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py

```diff
@@ -0,0 +1,272 @@
+        settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, GoogleAIChatPromptExecutionSettings)  # nosec
+
+        async_generator = self._send_chat_streaming_request(chat_history, settings)
+
+        async for messages in async_generator:
+            yield messages
+
+    async def _send_chat_streaming_request(
+        self,
+        chat_history: ChatHistory,
+        settings: GoogleAIChatPromptExecutionSettings,
+    ) -> AsyncGenerator[list[StreamingChatMessageContent], Any]:
+        """Send a streaming chat request to the Google AI service."""
+        genai.configure(api_key=self.service_settings.api_key.get_secret_value())
+        model = GenerativeModel(
+            self.service_settings.ai_model_id,
+            system_instruction=filter_first_system_message(chat_history),
+        )
+
+        response: AsyncGenerateContentResponse = await model.generate_content_async(
+            contents=self._prepare_chat_history_for_request(chat_history),
+            generation_config=GenerationConfig(**settings.prepare_settings_dict()),
+            stream=True,
+        )
+
+        async for chunk in response:
+            yield [self._create_streaming_chat_message_content(chunk, candidate) for candidate in chunk.candidates]
+
+    def _create_streaming_chat_message_content(
+        self,
+        chunk: GenerateContentResponse,
+        candidate: Candidate,
+    ) -> StreamingChatMessageContent:
+        """Create a streaming chat message content object.
+
+        Args:
+            chunk: The response from the service.
+            candidate: The candidate from the response.
+
+        Returns:
+            A streaming chat message content object.
+        """
+        # Best effort conversion of finish reason. The raw value will be available in metadata.
+        finish_reason: FinishReason | None = finish_reason_from_google_ai_to_semantic_kernel(candidate.finish_reason)
+        response_metadata = self._get_metadata_from_response(chunk)
+        response_metadata.update(self._get_metadata_from_candidate(candidate))
+
+        return StreamingChatMessageContent(
+            ai_model_id=self.ai_model_id,
+            role=AuthorRole.ASSISTANT,
+            choice_index=candidate.index,
+            content=candidate.content.parts[0].text,
+            inner_content=chunk,
+            finish_reason=finish_reason,
+            metadata=response_metadata,
+        )
+
+    # endregion
+
+    @override
+    def _prepare_chat_history_for_request(
+        self,
+        chat_history: ChatHistory,
+        role_key: str = "role",
+        content_key: str = "content",
+    ) -> list[Content]:
+        chat_request_messages: list[Content] = []
+
+        for message in chat_history.messages:
+            if message.role == AuthorRole.SYSTEM:
+                # Skip system messages since they are not part of the chat request.
```

Comment:
Can you clarify what "they are not part of the chat request" means. Are they not supported by this AI service?

Reply:
The way the Google AI consumes the system message is through the construction of the model object: https://ai.google.dev/gemini-api/docs/system-instructions?authuser=1&lang=python

Reply:
With the old Google Palm connector we had a way to map the role to the correct one that Google accepts. Why can't the `AuthorRole.SYSTEM` message be placed into the following:

```
model=genai.GenerativeModel(
  model_name="gemini-1.5-flash",
  system_instruction="You are a cat. Your name is Neko.")
```

https://ai.google.dev/gemini-api/docs/system-instructions?authuser=1&lang=python#examples

Reply:
Never mind. I see you do that in `_send_chat_request`.